templates/module.mk

```diff
@@ -43,6 +49,8 @@ BASE_TEMPLATES := \
 		$(DIR)/model_slha.hpp.in \
 		$(DIR)/observables.hpp.in \
 		$(DIR)/observables.cpp.in \
+		$(DIR)/one_loop_decay_diagrams.hpp.in \
```

Comment:
Can you please move `one_loop_decay_diagrams.{h,c}pp.in` to the `src/` directory? It appears to me that it is completely model-independent. (And we would avoid re-compiling it each time we re-generate a model.)

Reply:
The reason they're there is that `one_loop_decay_diagrams.hpp` includes `@ModelName@_decay_amplitudes.hpp`.

Reply:
I see. But why does it have to include it? Couldn't this inclusion be removed?

Reply:
If it is because of the `Decay_amplitude_FFV` structs, I think one can simply also move them to the `src/` directory. :)

Reply:
Because the only purpose of functions in `one_loop_decay_diagrams` is to create `Decay_amplitude` objects.

Reply:
Hehe. It's a russian nesting doll. `Decay_amplitude_FFV` include` `cxx_qft/ModelName_qft.hpp`.

Reply:
You are of course right. But non of this is model-specific, as far as I can see. If I understand correctly, you have nicely abstracted the different decay amplitudes in a model-generic way. So, this should go into `src/`.

Are you concerned that this file is then only included in one place? I think this is no problem at all. The benefit of moving it to `src/` would be that
* it is not compiled each time we are re-generating a model (which is currently very often).
* you reduce the overall binary size if you create more than one model, because the code ends up in `libflexisusy`
* we can make modifications to the code and don't trigger a longish re-generation of the model(s)

So, only benefits as far as I can see. :)

Reply:
A Matrjoschka? ;)

Then I'd propose to

* move the structs `Decay_amplitude_SSS` etc. to `src/decay_amplitudes.{hpp,cpp}`
* keep the rest of `templates/decay_amplitudes.hpp.in` as it is (it may include the Matrjoschka code).
* move `templates/one_loop_decay_diagrams.{hpp,cpp}.in` to `src/one_loop_decay_diagrams.{hpp,cpp}`
* => `src/one_loop_decay_diagrams.hpp` then includes `src/decay_amplitudes.hpp`.

Reply:
Done in 08a996b5ab1c63d667c38f2a3c9458ba1cd7e86c.

Reply:
Perfect, many thanks!